test/e2e/httpproxy/007_client_cert_auth_test.go

```diff
@@ -0,0 +1,423 @@
+			Namespace: namespace,
+			Name:      "echo-with-auth",
+		},
+		Spec: contourv1.HTTPProxySpec{
+			VirtualHost: &contourv1.VirtualHost{
+				Fqdn: "echo-with-auth.projectcontour.io",
+				TLS: &contourv1.TLS{
+					SecretName: "echo-with-auth",
+					ClientValidation: &contourv1.DownstreamValidation{
+						CACertificate: "echo-with-auth",
+					},
+				},
+			},
+			Routes: []contourv1.Route{
+				{
+					Services: []contourv1.Service{
+						{
+							Name: "echo-with-auth",
+							Port: 80,
+						},
+					},
+				},
+			},
+		},
+	}
+	fx.CreateHTTPProxyAndWaitFor(authProxy, httpProxyValid)
+
+	// This proxy requires a client certificate but does not verify it.
+	authSkipVerifyProxy := &contourv1.HTTPProxy{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      "echo-with-auth-skip-verify",
+		},
+		Spec: contourv1.HTTPProxySpec{
+			VirtualHost: &contourv1.VirtualHost{
+				Fqdn: "echo-with-auth-skip-verify.projectcontour.io",
+				TLS: &contourv1.TLS{
+					SecretName: "echo-with-auth-skip-verify",
+					ClientValidation: &contourv1.DownstreamValidation{
+						SkipClientCertValidation: true,
+					},
+				},
+			},
+			Routes: []contourv1.Route{
+				{
+					Services: []contourv1.Service{
+						{
+							Name: "echo-with-auth-skip-verify",
+							Port: 80,
+						},
+					},
+				},
+			},
+		},
+	}
+	fx.CreateHTTPProxyAndWaitFor(authSkipVerifyProxy, httpProxyValid)
+
+	// get the valid & invalid client certs
+	validClientCert := fx.Certs.GetTLSCertificate(namespace, "echo-client")
+	invalidClientCert := fx.Certs.GetTLSCertificate(namespace, "echo-client-invalid")
+
+	cases := map[string]struct {
+		host       string
+		clientCert *tls.Certificate
+		wantErr    string
+	}{
+		"echo-no-auth without a client cert should succeed": {
+			host:       noAuthProxy.Spec.VirtualHost.Fqdn,
+			clientCert: nil,
+			wantErr:    "",
+		},
+		"echo-no-auth with echo-client-cert should succeed": {
+			host:       noAuthProxy.Spec.VirtualHost.Fqdn,
+			clientCert: &validClientCert,
+			wantErr:    "",
+		},
+		"echo-no-auth with echo-client-cert-invalid should succeed": {
+			host:       noAuthProxy.Spec.VirtualHost.Fqdn,
+			clientCert: &invalidClientCert,
+			wantErr:    "",
+		},
+
+		"echo-with-auth without a client cert should error": {
+			host:       authProxy.Spec.VirtualHost.Fqdn,
+			clientCert: nil,
+			wantErr:    "tls: certificate required",
+		},
+		"echo-with-auth with echo-client-cert should succeed": {
+			host:       authProxy.Spec.VirtualHost.Fqdn,
+			clientCert: &validClientCert,
+			wantErr:    "",
+		},
+		"echo-with-auth with echo-client-cert-invalid should error": {
+			host:       authProxy.Spec.VirtualHost.Fqdn,
+			clientCert: &invalidClientCert,
+			wantErr:    "tls: certificate required",
+		},
+
+		"echo-with-auth-skip-verify without a client cert should succeed": {
+			host:       authSkipVerifyProxy.Spec.VirtualHost.Fqdn,
+			clientCert: nil,
+			wantErr:    "",
+		},
```

Comment:
@sunjayBhatia this is the case we were discussing.

Reply:
👍🏽 will update the other PR to cover this case